'test_bucket_update_labels' systest flakes with 400

From [this Kokoro failure](https://source.cloud.google.com/results/invocations/1a2dad90-74f8-4160-b0c0-e912f6a4fbe3/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-storage%2Fpresubmit%2Fsystem-3.8/log):

```python
__________________________ test_bucket_update_labels ___________________________

storage_client = <google.cloud.storage.client.Client object at 0x7fa25901f070>
buckets_to_delete = [<Bucket: update-labels-1629301504047>]

    def test_bucket_update_labels(storage_client, buckets_to_delete):
        bucket_name = _helpers.unique_name("update-labels")
        bucket = _helpers.retry_429_503(storage_client.create_bucket)(bucket_name)
        buckets_to_delete.append(bucket)
        assert bucket.exists()

        updated_labels = {"test-label": "label-value"}
        bucket.labels = updated_labels
        bucket.update()
        assert bucket.labels == updated_labels

        new_labels = {"another-label": "another-value"}
        bucket.labels = new_labels
        bucket.patch()
        assert bucket.labels == new_labels

        bucket.labels = {}
>       bucket.update()

tests/system/test_bucket.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
google/cloud/storage/bucket.py:949: in update
    super(Bucket, self).update(
google/cloud/storage/_helpers.py:394: in update
    api_response = client._put_resource(
google/cloud/storage/client.py:537: in _put_resource
    return self._connection.api_request(
google/cloud/storage/_http.py:78: in api_request
    return call()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <google.cloud.storage._http.Connection object at 0x7fa25901f310>
method = 'PUT', path = '/b/update-labels-1629301504047'
query_params = {'projection': 'full'}
data = '{"kind": "storage#bucket", "selfLink": "https://www.googleapis.com/storage/v1/b/update-labels-1629301504047", "id": "...cess": {"enabled": false}, "publicAccessPrevention": "unspecified"}, "locationType": "multi-region", "rpo": "DEFAULT"}'
content_type = 'application/json', headers = None, api_base_url = None
api_version = None, expect_json = True
_target_object = <Bucket: update-labels-1629301504047>, timeout = 60

    def api_request(
        self,
        method,
        path,
        query_params=None,
        data=None,
        content_type=None,
        headers=None,
        api_base_url=None,
        api_version=None,
        expect_json=True,
        _target_object=None,
        timeout=_DEFAULT_TIMEOUT,
    ):
        """Make a request over the HTTP transport to the API.

        You shouldn't need to use this method, but if you plan to
        interact with the API using these primitives, this is the
        correct one to use.

        :type method: str
        :param method: The HTTP method name (ie, ``GET``, ``POST``, etc).
                       Required.

        :type path: str
        :param path: The path to the resource (ie, ``'/b/bucket-name'``).
                     Required.

        :type query_params: dict or list
        :param query_params: A dictionary of keys and values (or list of
                             key-value pairs) to insert into the query
                             string of the URL.

        :type data: str
        :param data: The data to send as the body of the request. Default is
                     the empty string.

        :type content_type: str
        :param content_type: The proper MIME type of the data provided. Default
                             is None.

        :type headers: dict
        :param headers: extra HTTP headers to be sent with the request.

        :type api_base_url: str
        :param api_base_url: The base URL for the API endpoint.
                             Typically you won't have to provide this.
                             Default is the standard API base URL.

        :type api_version: str
        :param api_version: The version of the API to call.  Typically
                            you shouldn't provide this and instead use
                            the default for the library.  Default is the
                            latest API version supported by
                            google-cloud-python.

        :type expect_json: bool
        :param expect_json: If True, this method will try to parse the
                            response as JSON and raise an exception if
                            that cannot be done.  Default is True.

        :type _target_object: :class:`object`
        :param _target_object:
            (Optional) Protected argument to be used by library callers. This
            can allow custom behavior, for example, to defer an HTTP request
            and complete initialization of the object at a later time.

        :type timeout: float or tuple
        :param timeout: (optional) The amount of time, in seconds, to wait
            for the server response.

            Can also be passed as a tuple (connect_timeout, read_timeout).
            See :meth:`requests.Session.request` documentation for details.

        :raises ~google.cloud.exceptions.GoogleCloudError: if the response code
            is not 200 OK.
        :raises ValueError: if the response content type is not JSON.
        :rtype: dict or str
        :returns: The API response payload, either as a raw string or
                  a dictionary if the response is valid JSON.
        """
        url = self.build_api_url(
            path=path,
            query_params=query_params,
            api_base_url=api_base_url,
            api_version=api_version,
        )

        # Making the executive decision that any dictionary
        # data will be sent properly as JSON.
        if data and isinstance(data, dict):
            data = json.dumps(data)
            content_type = "application/json"

        response = self._make_request(
            method=method,
            url=url,
            data=data,
            content_type=content_type,
            headers=headers,
            target_object=_target_object,
            timeout=timeout,
        )

        if not 200 <= response.status_code < 300:
>           raise exceptions.from_http_response(response)
E           google.api_core.exceptions.BadRequest: 400 PUT https://storage.googleapis.com/storage/v1/b/update-labels-1629301504047?projection=full&prettyPrint=false: Invalid argument.

.nox/system-3-8/lib/python3.8/site-packages/google/cloud/_http.py:484: BadRequest
```

I can't reproduce locally.  @frankyn, @andrewsg Can either of you figure out why this test would get a 400 response?

--- a/tests/system/test_bucket.py
+++ b/tests/system/test_bucket.py
@@ -111,7 +111,9 @@ def test_bucket_update_labels(storage_client, buckets_to_delete):
     assert bucket.labels == new_labels
 
     bucket.labels = {}
-    bucket.update()
+    # See https://github.com/googleapis/python-storage/issues/541
+    retry_400 = _helpers.RetryErrors(exceptions.BadRequest)
+    retry_400(bucket.update)()
     assert bucket.labels == {}
 
 

